app/routes/companyInterest/components/CompanySemesterGUI.js

```diff
@@ -0,0 +1,124 @@
+  push: string => void,
+  events: Array<Object>,
+  semesters: Array<Object>,
+  autoFocus: any,
+  edit: boolean,
+  editSemester: (Array<Object>) => void
+};
+
+const CompanySemesterGUI = (props: Props) => {
+  const activeSemesters = semesters => (
+    <Flex column>
+      {semesters.map((semester, index) => (
+        <Flex key={index} className={styles.guiBoxes}>
+          <div className={styles.checkboxSpan}>{semesterToText(semester)}</div>
+          <Icon
+            name="close-circle"
+            onClick={() =>
+              props.editSemester({
+                ...semester,
+                activeInterestForm: false
+              })
+            }
+            className={styles.remove}
+          />
+        </Flex>
+      ))}
+    </Flex>
+  );
+
+  const { handleSubmit, autoFocus } = props;
+
+  return (
+    <Content>
+      <SemesterNavigation title="Endre aktive semestre" />
+      <Form onSubmit={handleSubmit}>
+        <Flex className={styles.guiWrapper}>
+          <Flex column style={{ marginRight: '50px' }}>
+            <label className={styles.heading}>Legg til aktivt semester</label>
+            <Field
+              autoFocus={autoFocus}
+              placeholder="2020"
+              label="År"
+              name="year"
+              type="number"
+              component={TextInput.Field}
+              className={styles.yearForm}
+              required
+            />
+            <RadioButtonGroup name="semester" label="Semester">
+              <Field
+                label="Vår"
+                component={RadioButton.Field}
+                inputValue="spring"
+              />
+              <Field
+                label="Høst"
+                component={RadioButton.Field}
+                inputValue="autumn"
+              />
+            </RadioButtonGroup>
+            <Button submit className={styles.submit}>
+              Legg til semester
+            </Button>
+          </Flex>
+          <Flex column>
+            <label className={styles.heading}>Deaktiver semestre</label>
+            {activeSemesters(props.activeSemesters)}
+          </Flex>
+        </Flex>
+      </Form>
+    </Content>
+  );
+};
+
+const onSubmit = ({ year, semester }, dispatch, props: Props) => {
+  const { semesters, addSemester, editSemester } = props;
+  const existingCompanySemester = semesters.find(companySemester => {
+    return (
+      companySemester.year == Number(year) &&
+      companySemester.semester == semester
+    );
+  });
+  if (existingCompanySemester)
+    return editSemester({
+      ...existingCompanySemester,
+      activeInterestForm: true
+    });
+  else return addSemester({ year, semester }); // Default is activeInterestForm: true
+};
+
+const validate = createValidator({
+  year: [required()],
+  semester: [required()]
+});
+
+export default legoForm({
+  form: 'addCompanySemester',
+  onSubmitSuccess: (result, dispatch) => dispatch(reset('addCompanySemester')),
```

Comment:
To avoid having to hardcode the form name twice, you can use this (example c): https://redux-form.com/6.0.0-alpha.4/docs/faq/howtoclear.md/

```jsx
submitMyForm(data) {
  const {createRecord, resetForm} = this.props;
  return createRecord(data).then(() => {
    resetForm();
    // do other success stuff
  });
}
```